examples/skywalking-tracing/docker-compose.yaml

```diff
@@ -0,0 +1,112 @@
+version: "3.7"
+services:
+
+  # Front envoy.
+  front-envoy:
+    build:
+      context: .
+      dockerfile: Dockerfile-frontenvoy
+    networks:
+      - envoymesh
+    expose:
+      - "8000"
+      - "8001"
+    ports:
+      - "8000:8000"
+      - "8001:8001"
+    depends_on:
+      - skywalking-oap
+  # First service.
+  service1:
+    build:
+      context: ../front-proxy
+      dockerfile: Dockerfile-service
+    volumes:
+      - ./service1-envoy-skywalking.yaml:/etc/service-envoy.yaml
+    networks:
+      - envoymesh
+    environment:
+      - SERVICE_NAME=1
+    expose:
+      - "8000"
+    depends_on:
+      - skywalking-oap
+  # Second service.
+  service2:
+    build:
+      context: ../front-proxy
+      dockerfile: Dockerfile-service
+    volumes:
+      - ./service2-envoy-skywalking.yaml:/etc/service-envoy.yaml
+    networks:
+      - envoymesh
+    environment:
+      - SERVICE_NAME=2
+    expose:
+      - "8000"
+    depends_on:
+      - skywalking-oap
+  # Skywalking components.
+  elasticsearch:
+    image: elasticsearch:7.9.2
+    container_name: elasticsearch
+    restart: always
+    networks:
+      - envoymesh
+    ports:
+      - 9200:9200
+    healthcheck:
+      test: ["CMD-SHELL", "curl --silent --fail localhost:9200/_cluster/health || exit 1"]
+      interval: 30s
+      timeout: 10s
+      retries: 3
+      start_period: 40s
+    environment:
+      discovery.type: single-node
+    expose:
+      - "9200"
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+  skywalking-oap:
+    image: apache/skywalking-oap-server:8.1.0-es7
+    container_name: oap
+    networks:
+      - envoymesh
+    depends_on:
```

Comment:
i think this needs to depend on `service_started` 

elasticsearch takes a while to start and if you try to access the admin ui immediately it fails (it works once everything is up and running)

as elasticsearch has a health check this should be quite straightforward

Reply:
actually admin ui fails - `curl http://localhost:8080` works

but then if you actually go to that address with a browser then it errors out  on the `graphql` request and the `ui` logs errors

```
Caused by: org.apache.http.conn.ConnectTimeoutException: Connect to skywalking-oap:12800 [skywalking-oap/172.28.0.4] failed: connect timed out
```
i think this may be to do with the  `container_name`

Reply:
ideally we need to test these calls - ie the ones that the ui makes - in `verify.sh`

Reply:
Yes. It does need to wait a while before it is truly ready.

Reply:
After that, skywalking ui will work normally.

Reply:
that should have been `service_healthy` rather than `service_started`

not sure - i think also removing `container_name` is required to make it work

Reply:
In fact,  i think container name will not affect it. I run this sandbox in my local compute and can get correct topography and tracing data from skywalking ui. It just need to wait.
But I will remove container_name. Anyway, this name make no sense. I copy this docker compose config from [skywalking community‘s examples](https://github.com/apache/skywalking-docker/blob/master/8/8.1.0/compose-es7/docker-compose.yml) and didn't make any changes.

Reply:
ive just realized that `docker-compose` versions `3+` dont support `depends_on.condition`

either we need to set the compose version to <3 or we need to wait for postgres another way - eg with `DELAY` in `verify.sh`

Reply:
But depends_on.condition just work in my computer. May be it's too old.

Reply:
Look like the CI can also run this docker-compose.yaml with depends_on.condition correctly. Anyway, i will remove it. According to the official documentation, denpends_on.condition is indeed no longer supported.

Reply:
wierd - yep - shouldnt work - https://stackoverflow.com/questions/47710767/what-is-the-alternative-to-condition-form-of-depends-on-in-docker-compose-version

and doesnt work for me locally - maybe docker_compose version i guess (even tho that shouldnt matter 8/)